homeassistant/components/philips_js/config_flow.py

```diff
@@ -53,34 +67,97 @@ async def async_step_import(self, conf: Dict[str, Any]):
             }
         )
 
+    async def _async_create_current(self):
+
+        system = self._current[CONF_SYSTEM]
+        return self.async_create_entry(
+            title=f"{system['name']} ({system.get('serialnumber', )})",
+            data=self._current,
+        )
+
+    async def async_step_pair(self, user_input: Optional[Dict] = None):
+        """Attempt to pair with device."""
+        assert self._hub
+
+        errors = {}
+
+        if user_input:
+            try:
+                username, password = await self._hub.pairGrant(
+                    self._pair_state, user_input[CONF_PIN]
+                )
+            except PairingFailure as exc:
+                LOGGER.debug(str(exc))
+                if exc.data.get("error_id") == "INVALID_PIN":
+                    errors[CONF_PIN] = "invalid_pin"
```

Comment:
This code is hard to read. From the whole `if user_input` statement, this is the case that will actually need the form rendered again. Would probably be easier to just return `async_show_form` here. 

And then rewrite it to be `if not user_input: return form` too.

Reply:
I always find config flow steps hard to read :). But I agree this ended up nicer.